### ISSUE TITLE
Travis builds started to fail on django1.4 (Ubuntu 12.04 LTS test failures)

### ISSUE THREAD

**Issue by swistakm** [bread-and-pepper/django-userena#480]
They do not fail on `tox==1.8.1`.

Travis build: https://travis-ci.org/bread-and-pepper/django-userena/jobs/63635501

Failures:
```
======================================================================
ERROR: test_signin_form (userena.tests.tests_forms.AuthenticationFormTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/bread-and-pepper/django-userena/userena/tests/tests_forms.py", line 107, in test_signin_form
    with override('en'):
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/__init__.py", line 118, in __enter__
    activate(self.language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/__init__.py", line 105, in activate
    return _trans.activate(language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 194, in activate
    _active.value = translation(language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 184, in translation
    current_translation = _fetch(language, fallback=default_translation)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 147, in _fetch
    res._info = res._info.copy()
AttributeError: 'NoneType' object has no attribute '_info'
======================================================================
ERROR: test_change_email_form (userena.tests.tests_forms.ChangeEmailFormTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/bread-and-pepper/django-userena/userena/tests/tests_forms.py", line 197, in test_change_email_form
    with override('en'):
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/__init__.py", line 118, in __enter__
    activate(self.language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/__init__.py", line 105, in activate
    return _trans.activate(language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 194, in activate
    _active.value = translation(language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 184, in translation
    current_translation = _fetch(language, fallback=default_translation)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 147, in _fetch
    res._info = res._info.copy()
AttributeError: 'NoneType' object has no attribute '_info'
======================================================================
ERROR: test_signup_form (userena.tests.tests_forms.SignupFormTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/bread-and-pepper/django-userena/userena/tests/tests_forms.py", line 67, in test_signup_form
    with override('en'):
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/__init__.py", line 118, in __enter__
    activate(self.language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/__init__.py", line 105, in activate
    return _trans.activate(language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 194, in activate
    _active.value = translation(language)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 184, in translation
    current_translation = _fetch(language, fallback=default_translation)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/translation/trans_real.py", line 147, in _fetch
    res._info = res._info.copy()
AttributeError: 'NoneType' object has no attribute '_info'
======================================================================
ERROR: test_preference_user (userena.tests.tests_middleware.UserenaLocaleMiddlewareTests)
Test the language preference of two users
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/bread-and-pepper/django-userena/userena/tests/tests_middleware.py", line 62, in test_preference_user
    self.failUnlessEqual(req.LANGUAGE_CODE, lang)
AttributeError: 'HttpRequest' object has no attribute 'LANGUAGE_CODE'
======================================================================
ERROR: test_password_reset_confirm (userena.tests.tests_views.UserenaViewsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/bread-and-pepper/django-userena/userena/tests/tests_views.py", line 470, in test_password_reset_confirm
    data={'email': 'john@example.com',})
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/test/client.py", line 449, in post
    response = super(Client, self).post(path, data=data, content_type=content_type, **extra)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/test/client.py", line 262, in post
    return self.request(**r)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/core/handlers/base.py", line 109, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/decorators.py", line 91, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/contrib/auth/views.py", line 153, in password_reset
    form.save(**opts)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/contrib/auth/forms.py", line 238, in save
    subject = loader.render_to_string(subject_template_name, c)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/template/loader.py", line 169, in render_to_string
    t = get_template(template_name)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/template/loader.py", line 145, in get_template
    template, origin = find_template(template_name)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/template/loader.py", line 138, in find_template
    raise TemplateDoesNotExist(name)
TemplateDoesNotExist: registration/password_reset_subject.txt
======================================================================
ERROR: test_password_reset_view_success (userena.tests.tests_views.UserenaViewsTests)
A ``POST`` to the password reset view with email that exists
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/bread-and-pepper/django-userena/userena/tests/tests_views.py", line 452, in test_password_reset_view_success
    data={'email': 'john@example.com',})
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/test/client.py", line 449, in post
    response = super(Client, self).post(path, data=data, content_type=content_type, **extra)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/test/client.py", line 262, in post
    return self.request(**r)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/core/handlers/base.py", line 109, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/utils/decorators.py", line 91, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/contrib/auth/views.py", line 153, in password_reset
    form.save(**opts)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/contrib/auth/forms.py", line 238, in save
    subject = loader.render_to_string(subject_template_name, c)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/template/loader.py", line 169, in render_to_string
    t = get_template(template_name)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/template/loader.py", line 145, in get_template
    template, origin = find_template(template_name)
  File "/home/travis/build/bread-and-pepper/django-userena/.tox/py26-django14/lib/python2.6/site-packages/django/template/loader.py", line 138, in find_template
    raise TemplateDoesNotExist(name)
TemplateDoesNotExist: registration/password_reset_subject.txt
----------------------------------------------------------------------
```

Investigate or simply stick to old tox version on order to be able to accept new PRs without problems.

**Comment by swistakm** [comment]
I was wrong, reason for why it fails is still unknown.

**Comment by swistakm** [comment]
Little success: I have managed to reproduce this issue in vagrant on Ubuntu 12.04 LTS (`hashicorp/precise64` image). So it seems to be system dependent issue.

**Comment by mbferg** [comment]
Encountering the same strange issue with translations in Django as of recently, also using 12.04 LTS.  Would like to know if you reach a resolution for this issue.

**Comment by swistakm** [comment]
It took me hours to find out that this is system dependent. I will take care of that in near future because it blocks our CI but I can't provide any guarantees when it will happen.

**Comment by SpectralAngel** [comment]
It is exactly as @mbferg pointed, some of the errors are a django issue with unavailable translations, documented here https://code.djangoproject.com/ticket/18192, I do not know if there a fix for 1.4.

**Comment by SpectralAngel** [comment]
Take a look at the resolution of django-recurrence, it just changed the LANGUAGE_CODE to something the application had defined in both the test app and the main app, it might actually fix the travis build. I noticed you have 'nl' defined in the locale directory of both the main userena app and its demo app.

**Comment by swistakm** [comment]
@SpectralAngel: Thanks for your insight but it did not help me to find the solution.

It appears that ``pip>7.0`` release messes up django installation in this environment and both django templates and locales are are installed in wrong place. This is too complex for me to find why. I will just  downgrade pip to ``6.1.1`` during travis builds. Hope it will fix that issue.

This seems to be the same issue as mentioned by someone on Stack Overflow:
http://stackoverflow.com/questions/30980682/djangos-locale-files-installed-in-weird-place

**Comment by swistakm** [comment]
fixed by #497
